crypto/fipsmodule/sha/sha3.c

```diff
@@ -332,16 +332,51 @@ int SHAKE_Squeeze(uint8_t *md, KECCAK1600_CTX *ctx, size_t len) {
     return 0;
   }
 
+
+
   // Skip FIPS202_Finalize if the input has been padded and
   // the last block has been processed
   if (ctx->state == KECCAK1600_STATE_ABSORB) {
     if (FIPS202_Finalize(md, ctx) == 0) {
       return 0;
     }
   }
+  // Process previous data from output buffer if any
+  if (ctx->buf_load != 0) {
+    if (len <= ctx->buf_load) {
+      OPENSSL_memcpy(md, ctx->buf + ctx->block_size - ctx->buf_load, len);
+      md += len;
+      ctx->buf_load -= len;
+      len = 0;
+      return 1;
+    } else {
+      OPENSSL_memcpy(md, ctx->buf + ctx->block_size - ctx->buf_load, ctx->buf_load);
+      md += ctx->buf_load;
+      len -= ctx->buf_load;
+      ctx->buf_load = 0;
+    }
+  }
 
-  Keccak1600_Squeeze(ctx->A, md, len, ctx->block_size, ctx->state);
-  ctx->state = KECCAK1600_STATE_SQUEEZE;
+  // Process all full size output requested blocks
+  block_bytes = ctx->block_size * (len / ctx->block_size);
+  if (len > ctx->block_size) {
+    Keccak1600_Squeeze(ctx->A, md, block_bytes, ctx->block_size, ctx->state);
+    md += block_bytes;
+    len -= block_bytes;
+    ctx->state = KECCAK1600_STATE_SQUEEZE;
+  }
```

Comment:
`block_bytes` could be declared and initialized inside the `if` block. It's not used elsewhere in this function.

Reply:
Thanks.